dask_glm/utils.py

```diff
@@ -205,3 +210,9 @@ def get_distributed_client():
         return get_client()
     except ValueError:
         return None
+
+
+def safe_zeros_like(X, shape):
+    if isinstance(X, da.Array):
+        return np.zeros_like(X._meta, shape=shape)
+    return np.zeros_like(X, shape=shape)
```

Comment:
```suggestion
    return zeros_like_safe(meta_from_array(X))
```

Reply:
You'll also need to `from dask.array.utils import meta_from_array` at the top.

Reply:
Sorry for the late reply, I think I might misunderstand our other conversion. This PR intends to enable `dask-glm` to deal with `pure numpy arrays`. Please let me know if not so and `dask-glm` should only accept `dask arrays`.

https://github.com/dask/dask-glm/blob/7b2f85fe043eb29212755e67e33e3df553ed0e58/dask_glm/algorithms.py#L100-L101

Let's say the input `X` is a pure `numpy` or `cupy` array, not a dask array. `beta = np.zeros_like(X._meta)` will be an error. The `safe_zeros_like` (bad naming) I implemented will check if `X` is a `pure numpy/cupy array` or a `dask array` and **return a `pure numpy/cupy array`**.  In contrast, `da.utils.zeros_like_safe` **returns a `dask array`.**  In this case the beta should be a `pure numpy/cupy array`.

Let me know if this clears things up. Thank you!

Reply:
> The `safe_zeros_like` (bad naming) I implemented will check if `X` is a `pure numpy/cupy array` or a `dask array` and **return a `pure numpy/cupy array`**.

That's exactly what `meta_from_array` does. It will return an array of the type `_meta` has (i.e., _chunk type_), so if the input is a NumPy array or a Dask array backed by NumPy, the result is an empty `numpy.ndarray`, and if the input is a CuPy array or a Dask array backed by CuPy, the result is an empty `cupy.ndarray`.

> In contrast, `da.utils.zeros_like_safe` **returns a `dask array`.**

That isn't necessarily true, it will only return a Dask array if the reference array is a Dask array. Because we're getting the _underlying chunk type_ with `meta_from_array`, the resulting array will be either a NumPy or CuPy array.

Reply:
Aha, that works! I will make the changes.